services/streammanager/wunderkind/wunderkindmanager.go

```diff
@@ -0,0 +1,110 @@
+package wunderkind
+
+import (
+	"encoding/json"
+
+	"github.com/aws/aws-sdk-go/service/lambda"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/rudderlabs/rudder-go-kit/awsutil"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/cmd/rudder-cli/config"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+)
+
+const (
+	INVOCATION_TYPE         = "RequestResponse"
+	WUNDERKIND_REGION       = "WUNDERKIND_REGION"
+	WUNDERKIND_IAM_ROLE_ARN = "WUNDERKIND_IAM_ROLE_ARN"
+	WUNDERKIND_EXTERNAL_ID  = "WUNDERKIND_EXTERNAL_ID"
+	WUNDERKIND_LAMBDA       = "WUNDERKIND_LAMBDA"
+)
+
+var (
+	jsonfast = jsoniter.ConfigCompatibleWithStandardLibrary
+)
+
+type inputData struct {
+	Payload string `json:"payload"`
+}
+
+type WunderkindProducer struct {
+	client LambdaClient
+	logger logger.Logger
+}
+
+type LambdaClient interface {
+	Invoke(input *lambda.InvokeInput) (*lambda.InvokeOutput, error)
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*WunderkindProducer, error) {
+	sessionConfig := &awsutil.SessionConfig{
+		Region:        config.GetEnv(WUNDERKIND_REGION),
+		IAMRoleARN:    config.GetEnv(WUNDERKIND_IAM_ROLE_ARN),
+		ExternalID:    config.GetEnv(WUNDERKIND_EXTERNAL_ID),
+		RoleBasedAuth: true,
+	}
+	awsSession, err := awsutil.CreateSession(sessionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &WunderkindProducer{
+		client: lambda.New(awsSession),
+		logger: logger.NewLogger().Child("streammanager").Child("wunderkind"),
+	}, nil
+}
+
+// Produce creates a producer and send data to Lambda.
+func (producer *WunderkindProducer) Produce(jsonData json.RawMessage, destConfig interface{}) (int, string, string) {
```

Comment:
Generally, it's single letter or two-letter for received names. Can we rename all receivers with p.
```suggestion
func (p *WunderkindProducer) Produce(jsonData json.RawMessage, destConfig interface{}) (int, string, string) {
```

Reply:
addressed